File: services/hub-webui/src/lib/schemas/identity.ts
import { z } from 'zod';

export const tenantCreateSchema = z.object({
  tenant_id: z.string().min(1, 'Tenant ID is required'),
  name: z.string().min(1, 'Name is required'),
  domain: z.string().optional(),
  spiffe_trust_domain: z.string().optional(),
  config: z.record(z.unknown()).optional(),
});

export const teamCreateSchema = z.object({
  team_id: z.string().min(1, 'Team ID is required'),
  tenant_id: z.string().min(1, 'Tenant ID is required'),
  name: z.string().min(1, 'Name is required'),
  description: z.string().optional(),
});

export const spiffeEntrySchema = z.object({
  spiffe_id: z.string().min(1, 'SPIFFE ID is required').regex(/^spiffe:\/\//, 'Must start with spiffe://'),
  tenant_id: z.string().min(1, 'Tenant ID is required'),
  parent_id: z.string().optional(),
  selectors: z.record(z.unknown()).optional(),
  ttl: z.number().int().min(0).default(0),
  dns_names: z.array(z.string()).optional(),
});

export type TenantCreateInput = z.infer<typeof tenantCreateSchema>;
export type TeamCreateInput = z.infer<typeof teamCreateSchema>;
export type SpiffeEntryInput = z.infer<typeof spiffeEntrySchema>;

File: services/hub-api/api/schemas/identity.py
"""Pydantic schemas for tenant, team, and SPIFFE identity endpoints."""
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, ConfigDict


class TenantCreateRequest(BaseModel):
    model_config = ConfigDict(strict=True)

    tenant_id: str
    name: str
    domain: Optional[str] = None
    spiffe_trust_domain: Optional[str] = None
    config: Optional[dict] = None


class TeamCreateRequest(BaseModel):
    model_config = ConfigDict(strict=True)

    team_id: str
    tenant_id: str
    name: str
    description: Optional[str] = None


class SpiffeEntryRequest(BaseModel):
    model_config = ConfigDict(strict=True)

    spiffe_id: str
    tenant_id: str
    parent_id: Optional[str] = None
    selectors: Optional[dict] = None
    ttl: int = 0
    dns_names: Optional[list[str]] = None
